Problem: upgrade secp256k1 to 0.20 fails

```
error[E0432]: unresolved import `secp256k1::All`
503
 --> src/types/key.rs:4:17
504
  |
505
4 | use secp256k1::{All, Secp256k1};
506
  |                 ^^^ no `All` in the root
507

508
error[E0599]: no method named `as_dyn_error` found for reference `&secp256k1::Error` in the current scope
509
   --> src/error.rs:12:13
510
    |
511
12  |     #[error("secp error")]
512
    |             ^^^^^^^^^^^^ method not found in `&secp256k1::Error`
513
    | 
514
   ::: /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/secp256k1-0.20.1/src/lib.rs:511:1
515
    |
516
511 | pub enum Error {
517
    | --------------
518
    | |
519
    | doesn't satisfy `secp256k1::Error: std::error::Error`
520
    | doesn't satisfy `secp256k1::Error: thiserror::private::AsDynError`
521
    |
522
    = note: the method `as_dyn_error` exists but the following trait bounds were not satisfied:
523
            `secp256k1::Error: std::error::Error`
524
            which is required by `secp256k1::Error: thiserror::private::AsDynError`
525
            `&secp256k1::Error: std::error::Error`
526
            which is required by `&secp256k1::Error: thiserror::private::AsDynError`
527

528
error[E0599]: no function or associated item named `signing_only` found for struct `secp256k1::Secp256k1<_>` in the current scope
529
  --> src/key_service/private_key_service.rs:36:44
530
   |
531
36 |         let signer = secp256k1::Secp256k1::signing_only();
532
   |                                            ^^^^^^^^^^^^ function or associated item not found in `secp256k1::Secp256k1<_>`
533

534
error[E0308]: mismatched types
535
  --> src/types/key.rs:72:14
536
   |
537
72 |         Self(secret_key)
538
   |              ^^^^^^^^^^ expected struct `secp256k1::SecretKey`, found struct `hdwallet::secp256k1::SecretKey`
539
   |
540
   = note: perhaps two different versions of crate `secp256k1` are being used?
541

542
error: aborting due to 4 previous errors
543

544
Some errors have detailed explanations: E0308, E0432, E0599.
545
For more information about an error, try `rustc --explain E0308`.
546
error: could not compile `chainlib`
547
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9,7 +9,6 @@ autoexamples = false
 serde = { version = "1.0", features = ["derive"], optional = true }
 sorted-json = { version = "0.1", optional = true }
 serde_json = { version = "1.0", optional = true }
-secp256k1 = { version = "0.17", default-features = false, features = ["rand"] }
 tiny-bip39 = "0.8"
 anyhow = { version = "1.0", features = ["std"] }
 hdwallet = "0.2"

--- a/src/error.rs
+++ b/src/error.rs
@@ -1,4 +1,5 @@
 use crate::hd_wallet::mnemonic::MnemonicError;
+use hdwallet::secp256k1;
 use thiserror::Error;
 
 #[derive(Error, Debug)]

--- a/src/key_service/ledger_service.rs
+++ b/src/key_service/ledger_service.rs
@@ -1,8 +1,8 @@
 use crate::ledger_app::CryptoApp;
 use crate::ledger_app::PubkeyAddress;
 use async_trait::async_trait;
+use hdwallet::secp256k1::PublicKey as InnerPublicKey;
 use ledger_transport::APDUTransport;
-use secp256k1::PublicKey as InnerPublicKey;
 use std::sync::Arc;
 use zx_bip44::BIP44Path;
 

--- a/src/key_service/private_key_service.rs
+++ b/src/key_service/private_key_service.rs
@@ -6,7 +6,7 @@ use crate::key_service::KeyService;
 use async_trait::async_trait;
 use bitcoin_hashes::{ripemd160, sha256};
 use bitcoin_hashes::{Hash, HashEngine};
-use secp256k1::Message;
+use hdwallet::secp256k1::{Message, Secp256k1};
 use stdtx::address::{Address, ADDRESS_SIZE};
 
 /// stores private key
@@ -33,7 +33,7 @@ impl PrivateKeyService {
         engine.input(msg);
         let hash = sha256::Hash::from_engine(engine);
         let message = Message::from_slice(hash.as_inner())?;
-        let signer = secp256k1::Secp256k1::signing_only();
+        let signer = Secp256k1::signing_only();
         let signature = signer.sign(&message, self.private_key.as_ref());
         let raw = signature.serialize_compact();
         let signature_str = base64::encode(&raw);

--- a/src/types/key.rs
+++ b/src/types/key.rs
@@ -1,6 +1,6 @@
 use crate::error::Error;
+use hdwallet::secp256k1;
 use hdwallet::ExtendedPrivKey;
-use secp256k1::rand::Rng;
 use secp256k1::{All, Secp256k1};
 use secp256k1::{Error as SecpError, PublicKey as InnerPublicKey, SecretKey};
 use serde::Serialize;
@@ -38,11 +38,6 @@ impl AsRef<SecretKey> for PrivateKey {
 }
 
 impl PrivateKey {
-    pub fn new<R: Rng + ?Sized>(rng: &mut R) -> Self {
-        let secret_key = SecretKey::new(rng);
-        Self(secret_key)
-    }
-
     pub fn from_slice(slice: &[u8]) -> Result<Self, SecpError> {
         let secret_key = SecretKey::from_slice(slice)?;
         Ok(Self(secret_key))

